File: Preproc/create_splits.pl
#!/usr/bin/perl -w

use strict;
use vars qw ();

use List::Util qw/shuffle/;

my $data_d = $ARGV[0];

if (not -e "$data_d/splits") {
    system("mkdir $data_d/splits");}

my @T = ("train","test");

foreach my $t (@T) {

    my $T = ucfirst($t);

    opendir(D,"$data_d/$T")||die;
    my @File = grep /\.conll/,readdir D;
    closedir(D);
    
    @File = shuffle @File;

    my $out_f = "$data_d/splits/$t\_dev.tsv";
    open(O,">$out_f")||die;
    my $i = 0;
    foreach my $f (@File) {
	$i++;
	if ($i > $#File / 2) {
	    $i = -100;
	    close(O);
	    open(O,">$data_d/splits/$t.tsv")||die;
	}
	print O "$f\n";
    }
    close(O);
}
system("mv $data_d/splits/test_dev.tsv $data_d/splits/devel.tsv");

File: Preproc_py/create_splits.py
import sys
import os,glob
import random

data_d = sys.argv[1]
#data_d = "/Users/david/Projects/Data/Experiments/Mutation/1"

if not os.path.isdir(data_d+"/splits"):
    os.mkdir(data_d+"/splits")

random.seed(30)
for t in ["train","test"]:
    T = t.capitalize()

    os.chdir(data_d+"/"+T)
    Files = glob.glob("*.conll")
    random.shuffle(Files)

    f = open(data_d+"/splits/"+t+"_dev.tsv",'w')
    i = 0
    for file in Files:
        i = i + 1
        if i > ((len(Files)-1) / 2):
            i = -100
            f.close()
            f = open(data_d+"/splits/"+t+".tsv",'w')
        f.write(file+"\n")
    f.close()

os.rename(data_d+"/splits/test_dev.tsv",data_d+"/splits/devel.tsv")

